app/gui2/shared/retry.ts

```diff
@@ -18,8 +18,9 @@ export interface BackoffOptions<E> {
   ) => boolean | void
   /** Called right before returning. */
   onSuccess?: (retryCount: number) => void
-  /** Called right before throwing an error. Note that `onBeforeRetry` is called for all tries
-   * before the last one. */
+  /** Called on the final retry, right before throwing an error.
+   * Note that `onBeforeRetry` is *not* called on the final retry, as there is nothing after the
```

Comment:
By `on the final retry` you mean `after final retry`? For me, `after` is more precise.